service/s3/unmarshal_error.go

```diff
@@ -48,6 +53,7 @@ func unmarshalError(r *request.Request) {
 	} else {
 		errCode = resp.Code
 		errMsg = resp.Message
+		err = nil
```

Comment:
so... it looks like `xml.Decoder.Decode` can actually return an `io.EOF` when the reader runs out of bytes, or was empty to begin with.